example/25-controllerregistration.yaml

```diff
@@ -13,6 +13,9 @@ spec:
   # autoEnable: # only valid if kind=Extension
   # - shoot
   # - seed
+  # clusterCompatibility: # only valid if kind=Extension
```

Comment:
> # only valid if kind=Extension

I was wondering if it would be generally helpful to specify this also for other extension kinds?

Reply:
Good question. 

My first impression was that this makes sense but it would mostly include some very confusing configurations, e.g. there can't be any `BackupEntry`, `ControlPlane`, `Infrastructure`, `Worker` with `clusterCompatibility: [garden,seed]` because the garden and seed reconcilers don't create those resources.
In contrast, a `BackupBucket` should actually never have `clusterCompatibility: [shoot]`.

We can consider `DNSRecord`s, although I think there is no benefit because the use-cases for DNS records is the same everywhere, if it's created for shoots, seeds, or the garden cluster.

In the end, we really need it for the `Extension` kind because it is very generic but for the rest it's confusing and almost inappropriate.

Reply:
I agree, it makes sense for now. We can probably lift the restriction to use it only for `kind=Extension` at any point later - if we really need it. Thank you.